include/Zydis/DecoderTypes.h

```diff
@@ -435,104 +435,115 @@ typedef ZyanU64 ZydisInstructionAttributes;
 typedef ZyanU32 ZydisCPUFlags;
 
 /**
- * Defines the `ZydisCPUFlag` enum.
+ * Defines the `ZydisFPUFlags` data-type.
  */
-typedef enum ZydisCPUFlag_
-{
-    /**
-     * Carry flag.
-     */
-    ZYDIS_CPUFLAG_CF,
-    /**
-     * Parity flag.
-     */
-    ZYDIS_CPUFLAG_PF,
-    /**
-     * Adjust flag.
-     */
-    ZYDIS_CPUFLAG_AF,
-    /**
-     * Zero flag.
-     */
-    ZYDIS_CPUFLAG_ZF,
-    /**
-     * Sign flag.
-     */
-    ZYDIS_CPUFLAG_SF,
-    /**
-     * Trap flag.
-     */
-    ZYDIS_CPUFLAG_TF,
-    /**
-     * Interrupt enable flag.
-     */
-    ZYDIS_CPUFLAG_IF,
-    /**
-     * Direction flag.
-     */
-    ZYDIS_CPUFLAG_DF,
-    /**
-     * Overflow flag.
-     */
-    ZYDIS_CPUFLAG_OF,
-    /**
-     * I/O privilege level flag.
-     */
-    ZYDIS_CPUFLAG_IOPL,
-    /**
-     * Nested task flag.
-     */
-    ZYDIS_CPUFLAG_NT,
-    /**
-     * Resume flag.
-     */
-    ZYDIS_CPUFLAG_RF,
-    /**
-     * Virtual 8086 mode flag.
-     */
-    ZYDIS_CPUFLAG_VM,
-    /**
-     * Alignment check.
-     */
-    ZYDIS_CPUFLAG_AC,
-    /**
-     * Virtual interrupt flag.
-     */
-    ZYDIS_CPUFLAG_VIF,
-    /**
-     * Virtual interrupt pending.
-     */
-    ZYDIS_CPUFLAG_VIP,
-    /**
-     * Able to use CPUID instruction.
-     */
-    ZYDIS_CPUFLAG_ID,
-    /**
-     * FPU condition-code flag 0.
-     */
-    ZYDIS_CPUFLAG_C0,
-    /**
-     * FPU condition-code flag 1.
-     */
-    ZYDIS_CPUFLAG_C1,
-    /**
-     * FPU condition-code flag 2.
-     */
-    ZYDIS_CPUFLAG_C2,
-    /**
-     * FPU condition-code flag 3.
-     */
-    ZYDIS_CPUFLAG_C3,
+typedef ZyanU8 ZydisFPUFlags;
 
-    /**
-     * Maximum value of this enum.
-     */
-    ZYDIS_CPUFLAG_MAX_VALUE = ZYDIS_CPUFLAG_C3,
-    /**
-     * The minimum number of bits required to represent all values of this enum.
-     */
-    ZYDIS_CPUFLAG_REQUIRED_BITS = ZYAN_BITS_TO_REPRESENT(ZYDIS_CPUFLAG_MAX_VALUE)
-} ZydisCPUFlag;
+/**
+ * Defines the `ZydisCPUFlag` data-type.
+ */
+typedef ZyanU8 ZydisCPUFlag;
+
+/**
+ * Carry flag.
+ */
+#define ZYDIS_CPUFLAG_CF     0
```

Comment:
Ok, I remembered that we also do this for all other things that are flags. However, the other flags (e.g instruction attributes) generally use the mask rather than the shift count as the value (e.g. `0x10` instead of 4). Should we also do this here?

Reply:
That as well came to my mind, but this is the only way to be backwards compatible.

I don't really mind having a breaking release this time tho. Effort to fix existing implementations is very minor for the users and at some point we have to remove the deprecated stuff anyways.

Reply:
Ah yeah, I see. Hmmm, that is unfortunate. Still, I don't want to break compat. We promise semantic versioning in the README and we should stick to that. We could release it as 4.0, but it kind of seems overkill. I'd say leave it as is.